### PR TITLE
Source Amazon Seller Partner: Add GET_SELLER_FEEDBACK_DATA report

### PR DESCRIPTION
**Pull request by lizdeika**
## What
Adds GET_SELLER_FEEDBACK_DATA report to Amazon Seller Partner source connector

## 🚨 User Impact 🚨
None

#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

### CODE REVIEW

**Comment by harshithmullapudi** [comment]
@lizdeika Hey can you move this PR out of Draft once you are done with changes. Thanks for your contribution.

**Comment by harshithmullapudi** [comment]
@lizdeika hey thanks for the contribution. I had a question
Can we use the `replication_start_date` and instead of a new key and then fetch it till date if it is full refresh and use the state for incremental sync.

**Comment by lizdeika** [comment]
> @lizdeika hey thanks for the contribution. I had a question
> Can we use the `replication_start_date` and instead of a new key and then fetch it till date if it is full refresh and use the state for incremental sync.

Hi,
`replication_start_date`(in this connector) maps to `createdSince`( in amazon documentation) which is for getting report itself that was created not earlier than `createdSince`.
`data_start_time` and `data_end_time` map to `dataStartTime` and `dataEndTime`. These are for the time period of a data in a report that we will be creating.
These 3 fields can not be reused between each other, they exist for different purposes
https://github.com/amzn/selling-partner-api-docs/blob/main/references/reports-api/reports_2021-06-30.md

**Comment by harshithmullapudi** [comment]
@lizdeika  thanks for sharing the info that was really helpful. Can you share across output for integrations test and unit tests ?

**Comment by lizdeika** [comment]
@harshithmullapudi I have educated myself on incremental syncs and slicing 
and rethought and remade the implementation as you suggested :)

**Comment by lizdeika** [comment]
@harshithmullapudi any idea why data from raw table(data gets synced successfully) does not go deduped _scd table when Sync mode is "Incremental | Deduped + history"

File: airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py [hunk 1]
```diff
@@ -341,6 +370,82 @@ class VendorInventoryHealthReports(ReportsAmazonSPStream):
...
+                "dataStartTime": start_date.strftime(DATE_TIME_FORMAT),
+                "dataEndTime": min(end_date_slice.subtract(seconds=1), end_date).strftime(DATE_TIME_FORMAT)
+            })
+            start_date = end_date_slice
+
+        return slices
+
+
+class SellerFeedbackReports(IncrementalReportsAmazonSPStream):
+    name = "GET_SELLER_FEEDBACK_DATA"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py:436]: Add doc string as other streams.

File: airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/schemas/GET_SELLER_FEEDBACK_DATA.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      "type": ["null", "string"]
+    },
+    "Response": {
+      "type": ["null", "string"]
+    },
+    "Order ID": {
+      "type": ["null", "string"]
+    },
+    "Rater Email": {
+      "type": ["null", "string"]
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/schemas/GET_SELLER_FEEDBACK_DATA.json:24]: Is it the name of this fields correct with the raw data/response from the API?

File: airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py [hunk 3]
```diff
@@ -9,32 +9,37 @@
...
 from Crypto.Cipher import AES
 from source_amazon_seller_partner.auth import AWSSignature
 
 REPORTS_API_VERSION = "2020-09-04"
 ORDERS_API_VERSION = "v0"
 VENDORS_API_VERSION = "v1"
 
-REPORTS_MAX_WAIT_SECONDS = 50
+# 33min. taken from real world experience working with amazon seller partner reports
+REPORTS_MAX_WAIT_SECONDS = 1980
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py:33]: This is based on your amount of data? Maybe transform this to a parameter where people can increase in necessity?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
small comments thanks for the contribution

File: airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py (hunk 1 above)
Comment (lizdeika) [reply airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py:436]: https://sellercentral.amazon.com/help/hub/reference/G202125660 this is the closest I can get

File: airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/schemas/GET_SELLER_FEEDBACK_DATA.json (hunk 2 above)
Comment (lizdeika) [reply airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/schemas/GET_SELLER_FEEDBACK_DATA.json:24]: Yes, fields are named exactly like this

File: airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py (hunk 3 above)
Comment (lizdeika) [reply airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py:33]: This is based on experience working with various different report types. They usually are generated faster but 30 minutes was maximum we saw. In any case 50 seconds is way to little.
Parameter is a good idea. Can I do it in another small PR?

Comment (marcosmarxm) [reply airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py:33]: sure.

**Comment by lizdeika** [comment]
@marcosmarxm any blocking stuff to get this in?

**Comment by lizdeika** [comment]
@marcosmarxm @harshithmullapudi any blocking stuff to get this in?

**Review by harshithmullapudi — APPROVED** [review]

**Pull request merged**